### PR TITLE
replace Travis CI workflow with GitHub Actions

### PR DESCRIPTION
**Pull request by astj**
This p-r intends to replace Travis CI workflows with GitHub Actions.
(I know we also have AppVeyor CI for Windows, but I'd like to keep it for now)

### CODE REVIEW

**Comment by astj** [comment]
demo: https://github.com/mackerelio/mackerel-agent/compare/github-actions...demo-github-actions?expand=1

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        go-version: ^1.14
+      id: go
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+    - run: make lint
+    - run: make convention
+    - run: make cover
+    - run: test `gofmt -l . | wc -l` = 0
+    - run: make crossbuild
+    #- run: goveralls -coverprofile=.profile.cov
```
Comment (astj) [inline .github/workflows/go.yml:25]: I'l try https://github.com/shogo82148/actions-goveralls

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: test
+    if: github.ref == 'refs/heads/master'
```
Comment (astj) [inline .github/workflows/go.yml:31]: will test this job by commenting out this line

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+    # before-deploy
+    - run: mkdir -p ~/rpmbuild/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
+    - run: docker pull mackerel/docker-mackerel-rpm-builder:c7
+    - run: docker pull mackerel/docker-mackerel-deb-builder
+    - run: make rpm deb rpm-kcps deb-kcps rpm-stage deb-stage tgz
+    - run: go get github.com/x-motemen/gobump/cmd/gobump
+    - run: go get github.com/mackerelio/golib/cmd/mackerel-github-release
+    # deploy
+    # - run: mackerel-github-release && mackerel-github-release -staging
```
Comment (astj) [inline .github/workflows/go.yml:44]: I should test only with `-staging`

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
+name: Go
+
+on: [push]
```
Comment (ne-sachirou) [inline .github/workflows/go.yml:3]: This is fired only when something is pushed to this repository. Pull Requests from other repository needs `pull_request` trigger.
https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request
```suggestion
on:
  pull_request: {}
  push:
    branches: [master]
```

**Pull request merged**